src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/GoogleServicesDAO.scala

```diff
@@ -28,6 +29,7 @@ trait GoogleServicesDAO extends ReportsSubsystemStatus {
   val fetchPriceList: Future[GooglePriceList]
   
   def writeObjectAsRawlsSA(bucketName: GcsBucketName, objectKey: GcsObjectName, objectContents: Array[Byte]): GcsPath
+  def writeObjectAsRawlsSA(bucketName: GcsBucketName, objectKey: GcsObjectName, tempFile: File): GcsPath
```

Comment:
new variation of `writeObjectAsRawlsSA` which accepts a temp file pointer as input instead of a materialized byte array